src/transformStateWithClones.js

```diff
@@ -6,8 +6,38 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const clones = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    const copyOfState = { ...currentState };
+
+    switch (action.type) {
```

Comment:
You can use a destruction for `action`
```suggestion
    const { type, .......} = action;
 
    switch (type) {
```